tests/test_layer.py

```diff
@@ -718,5 +721,153 @@ def helper_test_export_all_mount_points(layer):
+
+    # Empty modeling regions list
+    try:
+        layer.add_modeling_region("Tutorial Project", [])
+        pytest.fail("No exception raised for empty modeling regions list")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == "['Add modeling region error: Modeling regions list is empty.']"
+
+    # Invalid CCA name
+    invalid_region = copy.deepcopy(modeling_region_example)
+    invalid_region[0]["cca_name"] = ""
+    try:
+        layer.add_modeling_region("Tutorial Project", invalid_region)
+        pytest.fail("No exception raised for invalid CCA name")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == "['Add modeling region error: CCA name is invalid.']"
+
+    # Invalid region ID
+    invalid_region = copy.deepcopy(modeling_region_example)
+    invalid_region[0]["region_id"] = ""
+    try:
+        layer.add_modeling_region("Tutorial Project", invalid_region)
+        pytest.fail("No exception raised for invalid region ID")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == "['Add modeling region error: Region ID is invalid.']"
+
+    # Invalid region units
+    invalid_region = copy.deepcopy(modeling_region_example)
+    invalid_region[0]["region_units"] = ""
+    try:
+        layer.add_modeling_region("Tutorial Project", invalid_region)
+        pytest.fail("No exception raised for invalid region units")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == "['Add modeling region error: Region units are invalid.']"
+
+    # Missing shape
+    invalid_region = copy.deepcopy(modeling_region_example)
+    invalid_region[0].pop("shape")
+    try:
+        layer.add_modeling_region("Tutorial Project", invalid_region)
+        pytest.fail("No exception raised for missing shape")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == "['Add modeling region error: Shape is missing.']"
+
+    # Invalid shape type
+    invalid_region = copy.deepcopy(modeling_region_example)
+    invalid_region[0]["shape"] = "InvalidShapeType"
+    try:
+        layer.add_modeling_region("Tutorial Project", invalid_region)
+        pytest.fail("No exception raised for invalid shape type")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == "['Add modeling region error: Shape is not of a valid type.']"
+
+    # Invalid PCB model export type
+    invalid_region = copy.deepcopy(modeling_region_example)
+    invalid_region[0]["pcb_model_props"]["export_model_type"] = ""
+    try:
+        layer.add_modeling_region("Tutorial Project", invalid_region)
+        pytest.fail("No exception raised for invalid PCB model export type")
+    except SherlockAddModelingRegionError as e:
+        assert str(e.str_itr()) == (
+            "['Add modeling region error: PCB model export type is " "invalid.']"
```

Comment:
Please remove the extra set of double quotation marks before the word "invalid".